src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -576,21 +580,36 @@ public bool Execute(IConsole console, string command, params object[] inputs)
             if (ComplexCommand.AddLine(command, out fullCommand)
                 && !string.IsNullOrEmpty(fullCommand))
             {
+                // create a new token source with each command since CTS aren't usable once cancelled.
+                _tokenSource = new CancellationTokenSource();
+                _token = _tokenSource.Token;
                 return ExecuteHost(fullCommand, command, inputs);
             }
 
             return false; // constructing multi-line command
         }
 
-        protected static void OnExecuteCommandEnd()
+        protected void OnExecuteCommandEnd()
         {
             NuGetEventTrigger.Instance.TriggerEvent(NuGetEvent.PackageManagerConsoleCommandExecutionEnd);
+
+            // dispose token source related to this current command
+            _tokenSource?.Dispose();
```

Comment:
Its not shared state, i doubt one would ever use this token after command has completed execution but just to avoid any future mistake, I added this reset.

And about GetPathExpansionsAsyncCore then it simply replace current token with incoming one to make it work with tab expansion esc or cancel. And that extra step is just to set _token for host private data properties.